pageserver/src/tenant/config.rs

```diff
@@ -576,6 +553,38 @@ impl TryFrom<toml_edit::Item> for TenantConfOpt {
     }
 }
 
+/// This is a conversion from our internal tenant config object to the one used
+/// in external APIs.
+impl From<TenantConfOpt> for models::TenantConfig {
+    fn from(value: TenantConfOpt) -> Self {
+        fn humantime(d: Duration) -> String {
```

Comment:
For the sharding service, we can provide round-trip safety by using the same string-ization on the sharding service and in this function.  `FormattedDuration::fmt` is quite heavyweight (calculates all the years/months/days/secs/millis), so I'd rather avoid it in this context, where we will be looping over all the tenants.

I've added a note in https://github.com/neondatabase/neon/issues/6342 to remember to check this when rolling out the sharding service: it will probably make sense to have some shared helper function for all the places we would like to make an unambiguous String representation of a duration.

Reply:
+1 on the shared helper function idea. We could for example encode it via decimals (pseudo floats because floats are cursed).

Reply:
Cool, I'll pick that up when implementing the other side of this in #6342.